osdk/src/commands/debug.rs

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use crate::commands::execute_run_command;
+use crate::config_manager::{DebugConfig, DebugRole, RunConfig};
+
+use crate::util::{get_cargo_metadata, get_current_crate_info, get_target_directory};
+use std::env;
+use std::path::PathBuf;
+use std::process::Command;
+
+pub fn execute_debug_command(config: &DebugConfig) {
+    let DebugConfig {
+        manifest,
+        role,
+        cargo_args,
+    } = config;
+
+    let profile = profile_adapter(&cargo_args.profile);
+    match role {
+        DebugRole::Client => {
+            let file_path = get_target_directory()
+                .join("x86_64-unknown-none")
+                .join(profile)
+                .join(bin_file_name());
+            println!("debug {:?}", file_path);
+
+            let mut gdb = Command::new("gdb");
+            gdb.args([
+                "-ex",
+                "target remote :1234",
+                "-ex",
+                format!("file {}", file_path.display()).as_str(),
+            ]);
+            gdb.status().unwrap();
+        }
+        DebugRole::Server => {
+            vsc::config(workspace_root(), profile);
+
+            env::set_var(
+                "RUSTFLAGS",
+                env::var("RUSTFLAGS").unwrap_or_default()
+                    + " -g -C relocation-model=static -Z relro-level=off",
```

Comment:
The "-C relocation-model=static -Z relro-level=off" flags are default for Asterinas now (Since, there is no KASLR, LOL). Adding these made them duplicated. And "-g" seems default for debug builds.

If there is indeed KASLR in the future we neither shouldn't change the relocation model here but to disable KASLR via kernel command line or a feature gate.

See https://github.com/asterinas/asterinas/blob/4eff679eacb2415b3e3077d6c0d261e742d87f23/osdk/src/commands/build/mod.rs#L127-L136. And [our discussions](https://github.com/asterinas/asterinas/pull/679#discussion_r1524660729)

Reply:
The duplicate RUSTFLAGS options were added in case of forgetting to disable relocation after the link failures are resolved. They are removed now.

Sometimes I debug on a release build. It is much faster than the one in debug profile, especially on a machine without kvm.